internal/rpcserver/router.go

```diff
@@ -986,6 +986,18 @@ func (server *routedBoltzServer) createReverseSwap(ctx context.Context, isAuto b
 		if err != nil {
 			return nil, fmt.Errorf("invalid claim address %s: %w", claimAddress, err)
 		}
+
+		swaps, err := server.database.QueryReverseSwaps(database.SwapQuery{
+			States: []boltzrpc.SwapState{boltzrpc.SwapState_PENDING},
+		})
```

Comment:
https://github.com/BoltzExchange/boltz-client/pull/497/files#r2192366502

That should really be done. Let SQLite do the filtering for you